File: Week_01/id_1/LeetCode_189_01.py
#
# @lc app=leetcode id=189 lang=python3
# [189] Rotate Array
# https://leetcode.com/problems/rotate-array/description/

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        nums[:] = nums[n-k:] + nums[:n-k]

File: Week_01/id_53/LeetCode_189_01.c
void rotate(int* nums, int numsSize, int k){ 
    int i, j, p, q, t;
    if(numsSize<=1 || (k = k%numsSize) == 0) return;

    p = numsSize, q = k;
    while( (t = p % q) != 0 )
        p = q, q = t;
    
    for (p=0; p<q; p++) {
        for(i=p, t=nums[p]; (j = (numsSize+i-k) % numsSize) != p; i=j)
            nums[i] = nums[j];
        nums[i] = t; 
    }
    return;
}
